python/tvm/autotvm/graph_tuner/base_graph_tuner.py

```diff
@@ -0,0 +1,508 @@
+            When TVM compiles device specific program such as CUDA,
+            we also need host(CPU) side code to interact with the driver
+            setup the dimensions and parameters correctly.
+            target_host is used to specify the host side codegen target.
+            By default, llvm is used if it is enabled,
+            otherwise a stackvm intepreter is used.
+
+        infer_layout : bool, optional
+            Whether to infer layout transformation time if it doesn't exist in records, instead
+            of benchmarking on target device.
+
+            This might bring performance loss comparing to benchmarking layout transformation.
+        """
+        self._logger.info("Start to benchmark layout transformation...")
+        if records is None and infer_layout:
+            raise RuntimeError("Requires some records to infer layout transformation time.")
+
+        if isinstance(records, str):
+            records = load_from_file(records)
+            if not records and infer_layout:
+                raise RuntimeError("Records must be non-empty to infer layout transformation time.")
+
+        if isinstance(records, str):
+            records = load_from_file(records)
+        num_flops, total_time = 0, 0
+        if records is not None:
+            for record in records:
+                ltf_wkl = record[0].task.workload
+                self._layout_transform_dict[ltf_wkl] = record
+                input_shape = ltf_wkl[1][1]
+                flops = 1
+                for i in input_shape:
+                    flops *= i
+                num_flops += flops
+                total_time += record[1].costs[0]
+        avg_time = 0
+        if total_time > 0:
+            avg_time = total_time / num_flops
+
+        args_list = []
+        def _fetch_args_callback(from_node_idx, to_node_idx, from_sch_idx,
+                                 to_sch_idx, args):
+            """Callback function to fetch layout transform args"""
+            in_layout, out_layout = args[1], args[2]
+            if in_layout != out_layout:
+                args_list.append(args)
+
+        self._iterate_layout_transform(_fetch_args_callback)
+
+        def _log_to_list(record_list):
+            """Callback to log result to a list."""
+            def _callback(_, inputs, results):
+                """Callback implementation"""
+                record_list.append((inputs[0], results[0]))
+            return _callback
+
+        builder = autotvm.LocalBuilder(n_parallel=1, build_func=build_func)
+        runner = autotvm.LocalRunner(number=min_exec_num, repeat=1, timeout=timeout)
+        if use_rpc:
+            if device_key is None:
+                raise RuntimeError("device_key need to be set to use rpc tracker mode.")
+            runner = autotvm.measure.RPCRunner(device_key, host, port, n_parallel=n_parallel,
+                                               number=min_exec_num, repeat=1,
```

Comment:
why repeat=1?

Reply:
As long as run number > 1000, the result should be accurate enough. We don't need multiple running passes.